### PR TITLE
Add -Werror flag to the build

### LINKED ISSUE #739: Fix warnings during './gradlew clean assemble'
$ ./gradlew clean assemble

> Configure project :
Inferred project: opentelemetry-java-instrumentation, version: 0.7.0-SNAPSHOT

> Task :auto-bootstrap:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\agent-bootstrap\src\main\java\io\opentelemetry\auto\bootstrap\instrumentation\decorator\OrmClientDecorator.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :instrumentation:jdbc:javaccTemplate
Java Compiler Compiler Version 4.2 (Parser Generator)
(type "javacc" with no arguments for help)
Reading from file C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\jdbc\src\main\javacc\SqlNormalizer.jj . . .
Warning: Output directory "C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\jdbc\build\generated\javacc\template\io\opentelemetry\auto\instrumentation\jdbc\normalizer" does not exist. Creating the directory.
Note: UNICODE_INPUT option is specified. Please make sure you create the parser/lexer using a Reader with the correct character encoding.
File "TokenMgrError.java" does not exist.  Will create one.
File "ParseException.java" does not exist.  Will create one.
File "Token.java" does not exist.  Will create one.
File "SimpleCharStream.java" does not exist.  Will create one.
Parser generated with 0 errors and 1 warnings.

> Task :auto-tooling:compileJava
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:apache-httpasyncclient-4.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\apache-httpasyncclient-4.0\src\main\java\io\opentelemetry\auto\instrumentation\apachehttpasyncclient\ApacheHttpAsyncClientInstrumentation.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :auto-bootstrap:javadoc
C:\git\heyams\opentelemetry-java-instrumentation\agent-bootstrap\src\main\java\io\opentelemetry\auto\bootstrap\instrumentation\decorator\HttpServerTracer.java:124: warning - Tag @link: reference not found: #attachServerContext(Context, STORAGE)
C:\git\heyams\opentelemetry-java-instrumentation\agent-bootstrap\src\main\java\io\opentelemetry\auto\bootstrap\instrumentation\decorator\HttpServerTracer.java:124: warning - Tag @link: reference not found: #attachServerContext(Context, STORAGE)
C:\git\heyams\opentelemetry-java-instrumentation\agent-bootstrap\src\main\java\io\opentelemetry\auto\bootstrap\instrumentation\decorator\HttpServerTracer.java:124: warning - Tag @link: reference not found: #attachServerContext(Context, STORAGE)
3 warnings

> Task :instrumentation:grizzly-client-1.9:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\grizzly-client-1.9\src\main\java\io\opentelemetry\auto\instrumentation\grizzly\client\ClientResponseAdvice.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:geode-1.4:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\geode-1.4\src\main\java\io\opentelemetry\auto\instrumentation\geode\GeodeDecorator.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :auto-tooling:javadoc
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\bytebuddy\AgentCachingPoolStrategy.java:46: warning - invalid usage of tag &
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\muzzle\ReferenceCreator.java:58: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Instrumenter.java:215: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Instrumenter.java:221: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Utils.java:68: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Utils.java:73: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Utils.java:59: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Utils.java:59: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Utils.java:68: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Utils.java:73: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Utils.java:68: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Utils.java:73: warning - invalid usage of tag >
C:\git\heyams\opentelemetry-java-instrumentation\agent-tooling\src\main\java\io\opentelemetry\auto\tooling\Utils.java:59: warning - invalid usage of tag >

> Task :instrumentation:rxjava-1.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\rxjava-1.0\src\main\java\io\opentelemetry\auto\instrumentation\rxjava\TracedOnSubscribe.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
No types were transformed during plugin execution

> Task :instrumentation:hystrix-1.4:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\hystrix-1.4\src\main\java\io\opentelemetry\auto\instrumentation\hystrix\HystrixInstrumentation.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :auto-tooling:javadoc
13 warnings

> Task :instrumentation:jdbc:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\jdbc\build\generated\javacc\template\io\opentelemetry\auto\instrumentation\jdbc\normalizer\SqlNormalizer.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:kafka-streams-0.11:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\kafka-streams-0.11\src\main\java\io\opentelemetry\auto\instrumentation\kafkastreams\KafkaStreamsSourceNodeRecordDeserializerInstrumentation.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :instrumentation:rmi:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\rmi\src\main\java\io\opentelemetry\auto\instrumentation\rmi\context\server\ContextDispatcher.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\rmi\src\main\java\io\opentelemetry\auto\instrumentation\rmi\context\ContextPayload.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:ratpack-1.4:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\ratpack-1.4\src\main\java\io\opentelemetry\auto\instrumentation\ratpack\ActionWrapper.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:rediscala-1.8:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\rediscala-1.8\src\main\java\io\opentelemetry\auto\instrumentation\rediscala\RediscalaClientDecorator.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :instrumentation:spymemcached-2.12:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\spymemcached-2.12\src\main\java\io\opentelemetry\auto\instrumentation\spymemcached\MemcacheClientDecorator.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :instrumentation:twilio-6.6:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\twilio-6.6\src\main\java\io\opentelemetry\auto\instrumentation\twilio\TwilioAsyncInstrumentation.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:couchbase:couchbase-2.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\couchbase\couchbase-2.0\src\main\java\io\opentelemetry\auto\instrumentation\couchbase\v2_0\CouchbaseClientDecorator.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:elasticsearch:elasticsearch-rest-common:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\elasticsearch\elasticsearch-rest-common\src\main\java\io\opentelemetry\auto\instrumentation\elasticsearch\rest\ElasticsearchRestClientDecorator.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
No types were transformed during plugin execution

> Task :instrumentation:hibernate:hibernate-common:compileJava
No types were transformed during plugin execution

> Task :instrumentation:jaxrs-client:jaxrs-client-1.1:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\jaxrs-client\jaxrs-client-1.1\src\main\java\io\opentelemetry\auto\instrumentation\jaxrsclient\v1_1\InjectAdapter.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:elasticsearch:elasticsearch-transport-common:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\elasticsearch\elasticsearch-transport-common\src\main\java\io\opentelemetry\auto\instrumentation\elasticsearch\transport\ElasticsearchTransportClientDecorator.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :instrumentation:jaxrs:jaxrs-2.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\jaxrs\jaxrs-2.0\src\main\java\io\opentelemetry\auto\instrumentation\jaxrs\v2_0\DefaultRequestContextInstrumentation.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:elasticsearch:elasticsearch-transport-common:compileJava
No types were transformed during plugin execution

> Task :instrumentation:hibernate:hibernate-3.3:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\hibernate\hibernate-3.3\src\main\java\io\opentelemetry\auto\instrumentation\hibernate\v3_3\SessionFactoryInstrumentation.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:jedis:jedis-1.4:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\jedis\jedis-1.4\src\main\java\io\opentelemetry\auto\instrumentation\jedis\v1_4\JedisClientDecorator.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :instrumentation:jedis:jedis-3.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\jedis\jedis-3.0\src\main\java\io\opentelemetry\auto\instrumentation\jedis\v3_0\JedisClientDecorator.java uses or overrides a deprecated API.
Note: Recompile with -Xlint:deprecation for details.

> Task :instrumentation:hibernate:hibernate-4.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\hibernate\hibernate-4.0\src\main\java\io\opentelemetry\auto\instrumentation\hibernate\v4_0\SessionFactoryInstrumentation.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:jaxrs-client:jaxrs-client-2.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\jaxrs-client\jaxrs-client-2.0\src\main\java\io\opentelemetry\auto\instrumentation\jaxrsclient\v2_0\InjectAdapter.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:lettuce:lettuce-5.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\lettuce\lettuce-5.0\src\main\java\io\opentelemetry\auto\instrumentation\lettuce\v5_0\rx\LettuceMonoCreationAdvice.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:lettuce:lettuce-5.1:compileJava
warning: unknown enum constant When.MAYBE
  reason: class file for javax.annotation.meta.When not found
warning: unknown enum constant When.MAYBE
  reason: class file for javax.annotation.meta.When not found
warning: unknown enum constant When.MAYBE
  reason: class file for javax.annotation.meta.When not found
3 warnings

> Task :instrumentation:log4j:log4j-2.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\log4j\log4j-2.0\src\main\java\io\opentelemetry\auto\instrumentation\log4j\v2_0\ThreadContextInstrumentation.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:mongo:mongo-common:compileJava
No types were transformed during plugin execution

> Task :instrumentation:netty:netty-4.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\netty\netty-4.0\src\main\java\io\opentelemetry\auto\instrumentation\netty\v4_0\AttributeKeys.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:netty:netty-4.1:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\netty\netty-4.1\src\main\java\io\opentelemetry\auto\instrumentation\netty\v4_1\AttributeKeys.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:play-ws:play-ws-common:compileJava
No types were transformed during plugin execution

> Task :instrumentation:play:play-2.6:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\play\play-2.6\src\main\java\io\opentelemetry\auto\instrumentation\play\v2_6\PlayDecorator.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:aws-sdk:aws-sdk-2.2:auto:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\aws-sdk\aws-sdk-2.2\auto\src\main\java\io\opentelemetry\auto\instrumentation\awssdk\v2_2\TracingExecutionInterceptor.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:play-ws:play-ws-1.0:compileJava
Note: C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\play-ws\play-ws-1.0\src\main\java\io\opentelemetry\auto\instrumentation\playws\v1_0\StreamedAsyncHandlerWrapper.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:play-ws:play-ws-2.0:compileJava
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :instrumentation:play-ws:play-ws-2.1:compileJava
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

> Task :testing-common:compileGroovy
warning: Implicitly compiled files were not subject to annotation processing.
  Use -implicit to specify a policy for implicit compilation.
Note: C:\git\heyams\opentelemetry-java-instrumentation\testing-common\src\main\groovy\io\opentelemetry\auto\test\utils\ExceptionUtils.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
1 warning
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass (file:/C:/Users/heya/.gradle/caches/modules-2/files-2.1/org.codehaus.groovy/groovy/2.5.11/4d8f25c5da08af46fb204fd493ec894699a3f4e8/groovy-2.5.11.jar) to method java.util.AbstractCollection.hugeCapacity(int)
WARNING: Please consider reporting this to the maintainers of org.codehaus.groovy.reflection.CachedClass
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release

> Task :instrumentation:vertx-3.0:javadoc
C:\git\heyams\opentelemetry-java-instrumentation\instrumentation\vertx-3.0\src\main\java\io\opentelemetry\auto\instrumentation\vertx\package-info.java:10: warning - Tag @link: reference not found: io.opentelemetry.auto.instrumentation.vertx.reactive.VertxRxInstrumentatio1 warning

> Task :instrumentation:lettuce:lettuce-5.1:javadoc
warning: unknown enum constant When.MAYBE
  reason: class file for javax.annotation.meta.When not found
1 warning

> Task :instrumentation-core:spring:spring-boot-autoconfigure:javadoc
C:\git\heyams\opentelemetry-java-instrumentation\instrumentation-core\spring\spring-boot-autoconfigure\src\main\java\io\opentelemetry\instrumentation\spring\autoconfigure\webmvc\WebMVCFilterAutoConfiguration.java:26: warning - Tag @link: reference not found: WebMVCFilter
C:\git\heyams\opentelemetry-java-instrumentation\instrumentation-core\spring\spring-boot-autoconfigure\src\main\java\io\opentelemetry\instrumentation\spring\autoconfigure\webmvc\WebMVCFilterAutoConfiguration.java:26: warning - Tag @link: reference not found: WebMVCFilter
C:\git\heyams\opentelemetry-java-instrumentation\instrumentation-core\spring\spring-boot-autoconfigure\src\main\java\io\opentelemetry\instrumentation\spring\autoconfigure\webmvc\WebMVCFilterAutoConfiguration.java:26: warning - Tag @link: reference not found: WebMVCFilter
C:\git\heyams\opentelemetry-java-instrumentation\instrumentation-core\spring\spring-boot-autoconfigure\src\main\java\io\opentelemetry\instrumentation\spring\autoconfigure\webmvc\WebMVCFilterAutoConfiguration.java:26: warning - Tag @link: reference not found: WebMVCFilter
4 warnings

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD SUCCESSFUL in 5m 43s
825 actionable tasks: 797 executed, 28 from cache

### PR DESCRIPTION
**Pull request by trask**
Closes #739 (?)

### CODE REVIEW

**Review by anuraaga — APPROVED** [review]

**Comment by iNikem** [comment]
> Closes #739 (?)

Does it? I mean, have you tried and there are no more warnings? No javadoc warnings?

**Comment by trask** [comment]
> Does it? I mean, have you tried and there are no more warnings? No javadoc warnings?

Googled and added `-Xwerror` for javadoc task. Will review the errors it finds tomorrow. Anything else I should check?

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -24,7 +24,8 @@ jobs:
         run: .github/scripts/deadlock-detector.sh
 
       - name: Build
-        run: ./gradlew check --stacktrace -x :smoke-tests:test
+        # running javadoc to fail on javadoc warnings (using -Xwerror, see java.gradle)
+        run: ./gradlew check javadoc --stacktrace -x :smoke-tests:test
```
Comment (iNikem) [inline .github/workflows/pr.yaml:28]: Maybe just `build` then?

**Comment by trask** [comment]
@iNikem ready for another look

File: instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/OpenTelemetryService.java [hunk 2]
```diff
@@ -38,7 +38,7 @@ public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) throws Exc
           .whenComplete()
           .thenAccept(
               log -> {
-                if (HttpStatus.NOT_FOUND.equals(log.responseHeaders().status())) {
+                if (log.responseHeaders().status().equals(HttpStatus.NOT_FOUND)) {
```
Comment (iNikem) [inline instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/OpenTelemetryService.java]: This change is strange. Was there some warning about the order of comparison?

Comment (trask) [reply instrumentation/armeria-1.3/library/src/main/java/io/opentelemetry/instrumentation/armeria/v1_3/OpenTelemetryService.java]: merge from master, this change no longer exists in this PR

**Review by iNikem — APPROVED** [review]

**Pull request merged**